src/interpreter.rs

```diff
@@ -156,25 +155,25 @@ fn eval_inner(
 
                 Value::Symbol(symbol) if symbol == "lambda" => {
                     let cdr = list.cdr();
-                    let argnames = Rc::new(cdr.car()?.clone());
+                    let argnames = Rc::new(cdr.car()?);
                     let body = Rc::new(Value::List(cdr.cdr()));
 
                     Ok(Value::Lambda(Lambda {
-                        closure: env.clone(),
+                        closure: env,
                         argnames,
                         body,
                     }))
                 }
 
                 Value::Symbol(symbol) if symbol == "quote" => {
-                    let exp = list.cdr().car()?.clone();
+                    let exp = list.cdr().car()?;
```

Comment:
Just curious, did you have some sort of tool that helped you find all these unneeded Rc `.clone()`s? I suspected there were some but had no way to identify them all en masse

Reply:
`cargo clippy`, I use NeoVim with Ale to run this command every time I save a file

Reply:
Huh. My editor shows clippy hints inline, but I've never seen that particular rule